src/components/cylc/table/Table.vue

```diff
@@ -0,0 +1,247 @@
+              dense
+              flat
+              hide-details
+              multiple
+              outlined
+              placeholder="Filter by task state"
+              v-model="tasksFilter.states"
+            >
+              <template v-slot:item="slotProps">
+                <Task :status="slotProps.item.value"/>
+                <span class="ml-2">{{ slotProps.item.value }}</span>
+              </template>
+              <template v-slot:selection="slotProps">
+                <div class="mr-2" v-if="slotProps.index >= 0 && slotProps.index < maximumTasks">
+                  <Task :status="slotProps.item.value"/>
+                </div>
+                <span
+                  v-if="slotProps.index === maximumTasks"
+                  class="grey--text caption"
+                >
+            (+{{ tasksFilter.states.length - maximumTasks }})
+          </span>
+              </template>
+            </v-select>
+          </v-col>
+          <v-col
+            cols="12"
+            md="2">
+            <!-- TODO: we shouldn't need to set the height (px) here, but for some reason the Vuetify
+                       components don't seem to agree on the height here -->
+            <v-btn
+              id="c-table-filter-btn"
+              height="40"
+              block
+              outlined
+              @click="filterTasks"
+            >Filter</v-btn>
+          </v-col>
+        </v-row>
+      </v-col>
+    </v-row>
+    <table layout="auto" width="100%" padding="1px" border="1px solid black">
+      <thead>
+      <tr align="left">
+        <th>Task</th>
+        <th>State</th>
+        <th>Cyclepoint</th>
+        <th>Host</th>
+        <th>Job System</th>
+        <th>Job ID</th>
+        <th>T-submit</th>
+        <th>T-start</th>
+        <th>T-finish</th>
+        <th>dT-mean</th>
+      </tr>
+      </thead>
+      <tr
+          v-for="task of this.tasks"
+          :key="task.id"
+          >
+        <td>{{ task.id }}</td>
+        <td>{{ task.state }}</td>
+        <td>{{ task.cyclePoint }}</td>
+        <td>{{ getTaskJobProps(task, 'platform') }}</td>
+        <td>{{ getTaskJobProps(task, 'jobRunnerName') }}</td>
+        <td>{{ getTaskJobProps(task, 'jobId') }}</td>
+        <td>{{ getTaskJobProps(task, 'submittedTime') }}</td>
+        <td>{{ getTaskJobProps(task, 'startedTime') }}</td>
+        <td>{{ getTaskJobProps(task, 'finishedTime') }}</td>
+        <td>{{ task.meanElapsedTime }}</td>
+      </tr>
+    </table>
+  </v-container>
+</template>
+
+<script>
+import TaskState from '@/model/TaskState.model'
+import Task from '@/components/cylc/Task'
+import clonedeep from 'lodash.clonedeep'
+
+export default {
+  name: 'TableComponent',
+  props: {
+    tasks: {
+      type: Array,
+      required: true
+    },
+    hoverable: Boolean,
+    activable: Boolean,
+    multipleActive: Boolean,
+    filterable: {
+      type: Boolean,
+      default: true
+    }
+  },
+  components: {
+    Task
```

Comment:
You already have the `Task` component here. You should be able to import the `Job` component in the same manner, and declare it here (so that you can use it in the `template` section).

Then, in the `template` section, use these components to display the icon per task proxy, something like `<Task :status="taskProxy.state" />`

Reply:
`Job` component imported and declared.
Icons not yet displaying as will do this later.

Reply:
@kinow | re: icons,  Maybe I have this wrong, but isn't the `state` of a task, whether it is `running`, `waiting`, `preparing` etc?
I tried both `state` and `status` and at best I can get the state of a task to be replicated in the task name column. Looked at the tree component and the only icons I could find where associated with `+` and `-`, presumably having to do with the filtering section up the top. Where am I going wrong?

Reply:
>@kinow | re: icons, Maybe I have this wrong, but isn't the state of a task, whether it is running, waiting, preparing etc?

That's correct. If you have a `task` object, retrieved from our GraphQL API, then you will have a `task.state` with the state as you described above.

>I tried both state and status and at best I can get the state of a task to be replicated in the task name column. Looked at the tree component and the only icons I could find where associated with + and -, presumably having to do with the filtering section up the top. Where am I going wrong?

Here I'm talking about the [`prop`](https://github.com/cylc/cylc-ui/blob/2042834d65f00ca9a77db1836066c56397221d1d/src/components/cylc/Task.vue#L56) that `Task` and `Job` expect you to pass. Unfortunately it's not very consistent at the moment.

We have `state` in most places, but we still have `status` in other places (to be fixed soon-ish). So in this case, you should add to the Table the Task and the Job icons. Something like `<Task :status="taskObject.state">`.

Does this make sense?

Reply:
Task import and icons now workingf